src/frontEnd.js

```diff
@@ -0,0 +1,96 @@
+var game = null;
+
+$(document).ready(function() {
+  $("#start-game").click(function() {
+    event.preventDefault();
+    if (game !== null) {
+      alert("Game in progress. Click exit game to terminate.");
+      return;
+    }
+    let $playerName = $("#player-name").val();
+    $(".player-name").text($playerName);
+    game = new Game($playerName, Frame);
+    $("#player-name").val("");
+  });
+
+  $(".roll-button").click(function() {
+    if (game === null) {
+      alert("Please fill out your name and click start game");
+    }
+    let pinsKnocked = parseInt($(this).attr("name"));
+    game.roll(pinsKnocked);
+    let frameNumber = game.frameNumber;
+    let thisFrame = game.frames[frameNumber - 1];
+    let score = game.score;
+    if(thisFrame.rolls.length === 1) {
```

Comment:
It feels like these if statements could be simplified. What we essentially want to do, once we have handled the game mechanics stuff above and extracted the score, is:

(1) find the relevant element
(2) set the text of this element to the roll
(3) update the total frame score (if the frame is complete)
(4) hide any buttons that are invalid for the next roll (if the frame is not complete)

If we rethink our HTML slightly and uses classes to denote element types, we can do (1) e.g by:

```
var rollElement = $('.frames')[frameNumber - 1].find('rolls')[thisFrame.rolls.length - 1);
```

Then the rest of this can be done in sequence without branching.